tests/test_detectAndMeasure.py

```diff
@@ -217,6 +265,261 @@ def _check_diaSource(self, refSources, diaSource, refIds=None,
+                self.assertFloatsAlmostEqual(diaSource["ip_diffim_DipoleFit_separation"], offset, rtol=0.1)
+            else:
+                raise ValueError("DiaSource with ID %s is not a dipole!", diaSource.getId())
+
+    def test_sky_sources(self):
+        """Add sky sources and check that they are sufficiently far from other
+        sources and have negligible flux.
+        """
+        noiseLevel = 1.
+        staticSeed = 1
+        transientSeed = 6
+        transientFluxLevel = 1000.
+        transientFluxRange = 1.5
+        fluxLevel = 500
+        kwargs = {"seed": staticSeed, "psfSize": 2.4, "fluxLevel": fluxLevel}
+        science, sources = makeTestImage(noiseLevel=noiseLevel, noiseSeed=6, **kwargs)
+        matchedTemplate, _ = makeTestImage(noiseLevel=noiseLevel/4, noiseSeed=7, **kwargs)
+        config = detectAndMeasure.DetectAndMeasureScoreTask.ConfigClass()
+        config.doApCorr = False
+        config.doMerge = False
+        config.doSkySources = True
+        config.doForcedMeasurement = False
+        config.skySources.nSources = 5
+        kernelWidth = np.max(science.psf.getKernel().getDimensions())//2
+        detectionTask = detectAndMeasure.DetectAndMeasureScoreTask(config=config)
+        transients, transientSources = makeTestImage(seed=transientSeed, psfSize=2.4,
+                                                     nSrc=10, fluxLevel=transientFluxLevel,
+                                                     fluxRange=transientFluxRange,
+                                                     noiseLevel=noiseLevel, noiseSeed=8)
+        difference = science.clone()
+        difference.maskedImage -= matchedTemplate.maskedImage
+        difference.maskedImage += transients.maskedImage
+        subtractTask = subtractImages.AlardLuptonPreconvolveSubtractTask()
+        scienceKernel = science.psf.getKernel()
+        score = subtractTask._convolveExposure(difference, scienceKernel, subtractTask.convolutionControl)
+        output = detectionTask.run(science, matchedTemplate, difference, score)
+        skySources = output.diaSources[output.diaSources["sky_source"]]
+        self.assertEqual(len(skySources), config.skySources.nSources)
+        for skySource in skySources:
+            # The sky sources should not be close to any other source
+            with self.assertRaises(AssertionError):
+                self._check_diaSource(transientSources, skySource, matchDistance=kernelWidth)
+            with self.assertRaises(AssertionError):
+                self._check_diaSource(sources, skySource, matchDistance=kernelWidth)
+            # The sky sources should have low flux levels.
+            self._check_diaSource(transientSources, skySource, matchDistance=1000, scale=0.,
+                                  atol=np.sqrt(transientFluxRange*transientFluxLevel))
+
+    def _check_diaSource(self, refSources, diaSource, refIds=None,
```

Comment:
Should this be in a parent class or a free function? This version looks identical to the one in `DetectAndMeasureTest`.